packages/components/src/DataList/DataList.types.ts

```diff
@@ -97,6 +97,16 @@ export interface DataListProps<T extends DataListObject> {
    * The callback function when the user scrolls to the bottom of the list.
    */
   readonly onLoadMore?: () => void;
+
+  /**
+   * Callback when an item checkbox is clicked.
+   */
+  readonly onSelect?: (items: T["id"][]) => void;
+
+  /**
+   * The list of Selected Item ids
+   */
+  readonly selected?: Array<DataListObject["id"]>;
```

Comment:
Does this need to be `T["id"]` as well?

Reply:
@darryltec I was running into an issue. See https://github.com/GetJobber/atlantis/pull/1529#discussion_r1324651923

Reply:
That's odd that one of the prop works but the other doesn't. 

Where are you seeing the error? I changed the example id from gql string to just map index numbers. Which changed the `T` id to a `number` instead of a string and it seems to work fine.

Reply:
I was seeing it in the tests in `DataList.test.tsx`

Reply:
oh, that's because of the spread. since it is just grabbing a `typeof DataList` it defaults to number or string then tries to pass it in. more of a test setup problem than a consumer problem

Reply:
@darryltec I was able to fix it in 9312b4d73eef3b2a4f60f234a4605152d7d85206